core/database/foxx/api/repo_router.js

```diff
@@ -739,13 +759,29 @@ router
                         subject_id = req.queryParams.subject;
                     else subject_id = g_lib.getUserFromClientID(req.queryParams.subject)._id;
 
-                    var result = g_tasks.taskInitAllocDelete(
-                        client,
-                        req.queryParams.repo,
-                        subject_id,
-                    );
+                    // Find the repository using the new type system
+                    var findResult = RepositoryOps.find(req.queryParams.repo);
+                    if (!findResult.ok) {
+                        throw [findResult.error.code, findResult.error.message];
+                    }
+
+                    var repository = findResult.value;
+
+                    // Check permissions
+                    var permResult = RepositoryOps.checkPermission(repository, client._id, "admin");
+                    if (!permResult.ok || !permResult.value) {
+                        throw g_lib.ERR_PERM_DENIED;
+                    }
+
+                    // Delete allocation using the new system
+                    var deleteResult = RepositoryOps.deleteAllocation(repository, subject_id);
```

Comment:
DeleteAllocation does not check anywhere that the subject actually exists.

The below is from the initial taskInitAllocDelete call.
```
if (!g_db._exists(a_subject_id))                                                                                                                                       throw [g_lib.ERR_NOT_FOUND, "Subject, '" + a_subject_id + "', does not exist"];          
```

Reply:
This function is missing from the branch when I check it out so hard to follow if it is checking.

```
g_tasks.repoAllocationDeleteTask
```

but is called in foxx/api/globus.js